tests/components/streams_bootstrap/test_producer_app.py

```diff
@@ -60,11 +60,49 @@ def producer_app(self) -> ProducerApp:
                 },
             },
         )
+        assert producer.is_cron_job is False
+        return producer
+
+    @pytest.fixture()
+    def producer_app_cron_job(self) -> ProducerApp:
+        producer = ProducerApp(
+            name=PRODUCER_APP_NAME,
+            **{
+                "version": "3.2.1",
+                "namespace": "test-namespace",
+                "values": {
+                    "image": "ProducerApp",
+                    "kafka": {"bootstrapServers": "fake-broker:9092"},
+                    "schedule": "0 12 * * *",
+                },
+                "clean_schemas": True,
+                "to": {
+                    "topics": {
+                        "producer-app-output-topic": TopicConfig(
+                            type=OutputTopicTypes.OUTPUT, partitions_count=10
+                        ),
+                    }
+                },
+            },
+        )
+        assert producer.is_cron_job is True
+        return producer
 
     @pytest.fixture(autouse=True)
     def empty_helm_get_values(self, mocker: MockerFixture) -> MagicMock:
         return mocker.patch.object(Helm, "get_values", return_value=None)
 
+    def test_helm_name_override(self, producer_app: ProducerApp):
+        assert len(producer_app.helm_name_override) == 63
+        assert producer_app.helm_name_override == PRODUCER_APP_HELM_NAME_OVERRIDE
+
+    def test_cron_job_helm_name_override(self, producer_app_cron_job: ProducerApp):
+        assert len(producer_app_cron_job.helm_name_override) == 52
```

Comment:
I think we have a const for this. How about we use that?

Reply:
not for this new one. The Helm release name is limited to 53 chars, but Kubernetes CronJob is limited to 52

Reply:
also where possible test assertions should be independent from the code